src/config.c

```diff
@@ -473,8 +473,16 @@ void loadServerConfigFromString(char *config) {
         } else if ((!strcasecmp(argv[0],"slaveof") ||
                     !strcasecmp(argv[0],"replicaof")) && argc == 3) {
             slaveof_linenum = linenum;
+            if (!strcasecmp(argv[1], "no") && !strcasecmp(argv[2], "one")) {
+                server.masterhost = NULL;
```

Comment:
```suggestion
                sdsfree(server.masterhost);
                server.masterhost = NULL;
```
possible memory leak when using multiple `replicaof` directives in the config file.